src/modules/navigator/mission_block.cpp

```diff
@@ -337,7 +337,10 @@ MissionBlock::is_mission_item_reached_or_completed()
 				acceptance_radius = _mission_item.acceptance_radius;
 			}
 
-			float alt_acc_rad_m = _navigator->get_altitude_acceptance_radius();
+			struct position_setpoint_s *curr_sp = &_navigator->get_position_setpoint_triplet()->current;
+
+			float alt_acc_rad_m = PX4_ISFINITE(curr_sp->alt_acceptance_radius) ? curr_sp->alt_acceptance_radius :
```

Comment:
Is there a reason you did not add this directly in the  _navigator->get_altitude_acceptance_radius function? Then you could use it for other items as well. I don't think it would hurt, no?